src/gradlecheck/FetchPostMergeFailedTestClass.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package gradlecheck
+
+import groovy.json.JsonOutput
+import gradlecheck.OpenSearchMetricsQuery
+
+class FetchPostMergeFailedTestClass {
+    String metricsUrl
+    String awsAccessKey
+    String awsSecretKey
+    String awsSessionToken
+    def script
+
+    FetchPostMergeFailedTestClass(String metricsUrl, String awsAccessKey, String awsSecretKey, String awsSessionToken, def script) {
```

Comment:
except markdown class, every class has the same constructor and `get_query` methods. Would it be better to have a parent/super class or interface, if supported in groovy class, and have these method defined there and other classes just inherit it and implement in its own way. Method like `getPostMergeFailedTestClass` can have a generic name and implemented in all the sub-classes. 
Not a blocker for me.

Reply:
That was my initial implementation rishab, take a look at this [commit](https://github.com/opensearch-project/opensearch-build-libraries/commit/b8462b4d890cc2a218bab88924f006d5a1ff0cbd#diff-2087681af514d648a0a6740a4b91375803a761294436f8d010934d4ea7ffdd8d) which has the `extends` and `super` implementation, the jenkins pipeline works but this  `extends` and `super` implementation ran into issues with running the tests (both old and new tests part of the PR) part of this repo, please check my [comment](https://github.com/opensearch-project/opensearch-build-libraries/pull/436#issuecomment-2162026372) here,